### PR TITLE
1.0.x/generators

### PR DESCRIPTION
**Pull request by jmanero-r7**
This PR removes hard-coded tasks for generating specific types of projects, replacing them with a collection of resources in the `project` collection. Resources'  `action` attributes supports the same `:rm`, `:ignore`, `:create`, and `:sync` verbs, but resources can now be defined and modified directly through the Config interface, rather than requiring a code change to the gem.

Using the new `generator.project.current` reference allows a number of other attributes to be defined per-project-type as well.

### CODE REVIEW

File: lib/builderator/config/defaults.rb [hunk 1]
```diff
@@ -95,33 +98,75 @@ module Config
         end
       end
 
-      generator.gemfile.vagrant do |vagrant|
-        vagrant.install true
-        vagrant.version 'v1.8.0'
-      end
+      generator.project :default do |base|
+        base.ruby.version '2.1.5'
```
Comment (athompson-r7) [inline lib/builderator/config/defaults.rb]: Is there a reason for this?  Are we only running the test suite against this version?  I've seen developers struggle with this constraint.

Comment (jmanero-r7) [inline lib/builderator/config/defaults.rb]: This is the version currently available on razor CI (/me _shrugs_)

Comment (jmanero-r7) [inline lib/builderator/config/defaults.rb]: We should actually leave this constraint out of the defaults as it's specific to out own CI...

**Comment by rvanburen-r7** [comment]
LGTM

File: lib/builderator/config/defaults.rb (hunk 1 above)
Comment (jmanero-r7) [inline lib/builderator/config/defaults.rb]: Heh, now that I'm looking at it, this value isn't even in the Gemfile template :grimacing: ...

**Comment by athompson-r7** [comment]
LGTM

**Pull request merged**